File: PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/settings/AzureSettingsConfigurable.java
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */

package com.microsoft.azure.toolkit.intellij.settings;

import com.intellij.application.options.OptionsContainingConfigurable;
import com.intellij.openapi.options.Configurable;
import com.intellij.openapi.options.ConfigurationException;
import com.intellij.openapi.options.SearchableConfigurable;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

import static com.microsoft.azure.toolkit.intellij.common.AzureBundle.message;

public class AzureSettingsConfigurable implements SearchableConfigurable, Configurable.NoScroll, OptionsContainingConfigurable {
    public static final String AZURE_CONFIGURABLE_ID = "com.microsoft.intellij.AzureConfigurable";

    private java.util.List<Configurable> myPanels;
    private final AzureSettingsPanel azureSettingsPanel;

    public AzureSettingsConfigurable() {
        this.azureSettingsPanel = new AzureSettingsPanel();
    }

    @NotNull
    @Override
    public String getId() {
        return AZURE_CONFIGURABLE_ID;
    }

    @Nls
    @Override
    public String getDisplayName() {
        return azureSettingsPanel.getDisplayName();
    }

    @Nullable
    @Override
    public String getHelpTopic() {
        return "windows_azure_project_properties";
    }

    @Override
    public JComponent createComponent() {
        azureSettingsPanel.init();
        return azureSettingsPanel.getPanel();
    }

    @Override
    public boolean isModified() {
        return azureSettingsPanel.isModified();
    }

    @Override
    public void apply() throws ConfigurationException {
        if (!azureSettingsPanel.doOKAction()) {
            throw new ConfigurationException(message("setPrefErMsg"), message("errTtl"));
        }
    }

    @Override
    public void reset() {
        azureSettingsPanel.reset();
    }
}

File: PluginsAndFeatures/azure-toolkit-for-rider/src/main/kotlin/com/microsoft/azure/toolkit/intellij/settings/AzureSettingsConfigurable.kt
package com.microsoft.azure.toolkit.intellij.settings

import com.intellij.application.options.OptionsContainingConfigurable
import com.intellij.openapi.options.Configurable
import com.intellij.openapi.options.SearchableConfigurable
import javax.swing.JComponent

class AzureSettingsConfigurable : SearchableConfigurable, Configurable.NoScroll, OptionsContainingConfigurable {
    companion object {
        private const val AZURE_CONFIGURABLE_ID = "com.microsoft.intellij.RiderAzureConfigurable"
        private const val DISPLAY_NAME = "Azure"
    }

    private val azureSettingsPanel = AzureSettingsPanel()

    override fun getDisplayName() = DISPLAY_NAME
    override fun getId() = AZURE_CONFIGURABLE_ID

    override fun createComponent(): JComponent {
        azureSettingsPanel.init()
        return azureSettingsPanel.getPanel()
    }

    override fun isModified() = azureSettingsPanel.isModified()

    override fun apply() {
        azureSettingsPanel.apply()
    }

    override fun reset() {
        azureSettingsPanel.reset()
    }
}